### DISCUSSION TITLE
Issue with WhatsApp Interactive Templates in Chatwoot

### DISCUSSION THREAD

**Discussion by anicolco** [chatwoot/chatwoot#9991] [Self hosted]
Hello Chatwoot Team,

First of all, I want to congratulate you on the fantastic project you’ve developed. Chatwoot is an impressive tool and has been incredibly useful for us.

I’m currently experiencing an issue with WhatsApp interactive templates in Chatwoot. Specifically, the feedback and catalog templates are not functioning correctly. Below are the details of the issue:

Issue Description
Platform: Chatwoot installed on Docker with the latest version (chatwoot/chatwoot:latest).
Main Issue: WhatsApp interactive templates are not being sent correctly.
Feedback Template:
This template involves a survey where the user is asked to rate from 1 to 5, with an optional comment field.
Problem: When selecting and sending this template from Chatwoot, it appears to be sent successfully, but the user does not receive anything.
Additional Info: If I send the same template via an HTTP request (bypassing Chatwoot), and the user responds, the response is not registered in Chatwoot.
Example of the JSON used in the HTTP request:

json

```
{
  "messaging_product": "whatsapp",
  "to": "number",
  "type": "template",
  "template": {
    "name": "feedback_zoom",
    "language": {
      "code": "en"
    },
    "components": [
      {
        "type": "body",
        "parameters": [
          {
            "type": "text",
            "text": "name"
          },
          {
            "type": "text",
            "text": "text2"
          }
        ]
      },
      {
        "type": "button",
        "sub_type": "flow",
        "index": "0",
        "parameters": [
          {
            "type": "payload",
            "payload": "FLOW_START"
          }
        ]
      }
    ]
  }
}
```
Webhook data received:

json
```

{
  "object": "whatsapp_business_account",
  "entry": [
    {
      "id": "xxxx",
      "changes": [
        {
          "value": {
            "messaging_product": "whatsapp",
            "metadata": {
              "display_phone_number": "phone",
              "phone_number_id": "phone"
            },
            "contacts": [
              {
                "profile": {
                  "name": "AYR"
                },
                "wa_id": "phone"
              }
            ],
            "messages": [
              {
                "context": {
                  "from": "phone",
                  "id": "wamid.xxx"
                },
                "from": "number",
                "id": "wamid.xxx=",
                "timestamp": "1724158653",
                "type": "interactive",
                "interactive": {
                  "type": "nfm_reply",
                  "nfm_reply": {
                    "response_json": "{\"flow_token\":\"unused\",\"screen_0_RadioButtonsGroup_0\":\"0_★★★★★_•_Excellent_(5\\/5)\",\"wa_flow_response_params\":{\"flow_id\":\"1057592196023663\",\"flow_name\":\"Message templates_UTILITY_52ec8242-2\",\"title\":\"Responder encuesta\"},\"screen_0_TextArea_1\":\"Opcional…..\"}",
                    "body": "Sent",
                    "name": "flow"
                  }
                }
              }
            ]
          },
          "field": "messages"
        }
      ]
    }
  ]
}
```
Catalog Template:
Problem: The same issue occurs as with the feedback template. The template appears to be sent, but the user does not receive anything.
Additional Info: When sending the catalog template directly via an HTTP request, it is sent successfully, but there is no interaction record in Chatwoot.
Example of the JSON used in the HTTP request:

json
```

{
  "messaging_product": "whatsapp",
  "to": "number",
  "type": "template",
  "template": {
    "name": "catlogo",
    "language": {
      "code": "es"
    },
    "components": [
      {
        "type": "button",
        "sub_type": "catalog",
        "index": "0"
      }
    ]
  }
}
```
Current Situation:
In the meantime, we are using another Meta Business WhatsApp app with a different webhook through n8n to send these templates. However, it would be ideal to manage everything directly through Chatwoot.

I hope this information helps in investigating and resolving the issue. I look forward to any suggestions or fixes you might have.

Thank you again for your excellent work.

Best regards,
Alex

**Comment by viniciusgushiken** [comment]
By "interactive templates" do you mean WhatsApp Flows?

**Reply by syeddsherazi** [reply]
Were you able to figure this out? We're facing the same issue
@anicolco

**Comment by anicolco** [comment]
Sorry for the delay in responding—I hadn’t seen your message.

I also noticed @syeddsherazi is experiencing the same issue. To clarify: our feedback template is one of the new WhatsApp Flows (you can see the "flow_token" and screen_0_* parameters in the webhook payload), whereas our catalog template is a classic interactive template ("sub_type": "catalog").

Since the Flow-based template appears to send successfully from Chatwoot but never reaches the user, it seems Chatwoot doesn’t yet support WhatsApp Flows. Do you have any plans to add Flow support soon, or is there a recommended workaround in the meantime?

Thanks again for any guidance!

**Comment by kkim212** [comment]
same issue, and even I setup agentic bot, the message sent by chatwoot does not contain nfm_reply which is a flow response, content is null, probably chatwoot removed the data when it process the message

maybe the last solution I can do is to setup a webhook server to connect to meta, then pass the message back to chatwoot.... or it should be a quick fix for the chatwoot team, just to keep the nfm_reply for the message to agentic bot, so at least we can handle the message in the agentic bot

**Comment by alejandromontoya-hr** [comment]
I solved it as follows: I created a second application on https://developers.facebook.com/ and linked it to the same organization. For each app, I created a separate webhook — one for Chatwoot and another for n8n. This way, when the organization receives a message, the same payload is sent to both webhooks. From that point on, it depends on your specific needs. In my case, I set up a filter in n8n to select only interactive messages, while the rest were sent to “no operation, do nothing.” As for Chatwoot, since it doesn’t capture that data, those messages simply don’t appear. It really depends on each person’s requirements and creativity. But yes — there is a solution, and it works well.

**Reply by ronaldquinoo** [reply]
Hermano, estoy enfrentando el mismo problema, necesito tu ayuda para que me guíes paso a paso con la solución, te lo agradecería demasiado.

**Reply by mateusimao** [reply]
@alejandro-montoya-sagy também estou com a mesma necessidade, ter as respostas do flow pelo webhook recebido no chatwoot. No entanto, o webhook é definido para a conta WA, não entendi como fez para adicionar 2 webhooks para o mesmo whatsapp. Poderia nos informar?

**Reply by alejandromontoya-hr** [reply]
Los mensajes de WhatsApp API llegan a todas las aplicaciones de la organización, y estas se encargan de enrutar los mensajes a los webhooks que tengas configurados. Esto es algo normal y esperado.

Podemos aprovechar esto para crear dos aplicaciones: una que envíe los **MENSAJES NO INTERACTIVOS** a Chatwoot (te comparto este video para que configures tu Chatwoot con n8n: https://youtu.be/3yfBB4YdPNI?si=8ix1YMT7TUDibKPo) y otra que reciba los **MENSAJES INTERACTIVOS**, como los flows; en mi caso, también lo implementé con n8n.

A partir de aquí, puedes ajustarlo según tus necesidades.
